src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostAddressesTest.cs

```diff
@@ -118,5 +139,43 @@ public void DnsGetHostAddresses_LocalHost_ReturnsSameAsGetHostEntry()
 
             Assert.Equal(ipEntry.AddressList, addresses);
         }
+
+        [Theory]
+        [InlineData(false, AddressFamily.InterNetwork)]
+        [InlineData(false, AddressFamily.InterNetworkV6)]
+        [InlineData(true, AddressFamily.InterNetwork)]
+        [InlineData(true, AddressFamily.InterNetworkV6)]
+        public void DnsGetHostAddresses_LocalHost_AddressFamilySpecific(bool useAsync, AddressFamily addressFamily)
+        {
+            IPAddress[] addresses =
+                useAsync ? Dns.GetHostAddressesAsync(TestSettings.LocalHost, addressFamily).GetAwaiter().GetResult() :
+                Dns.GetHostAddresses(TestSettings.LocalHost, addressFamily);
```

Comment:
Why blocking rather than:
```suggestion
            IPAddress[] addresses =
                useAsync ? await Dns.GetHostAddressesAsync(TestSettings.LocalHost, addressFamily) :
                Dns.GetHostAddresses(TestSettings.LocalHost, addressFamily);
```
?